agent/handlers/v2/response.go

```diff
@@ -272,3 +288,34 @@ func NewContainerResponse(
 	resp.Volumes = v1.NewVolumesResponse(dockerContainer)
 	return resp
 }
+
+// metadataErrorHandling writes an error to the logger, and append an error response
+// to V4 metadata endpoint task response
+func metadataErrorHandling(resp *TaskResponse, err error, field, resourceARN string, includeV4Metadata bool) {
+	seelog.Errorf("V2 response: unable to get '%s' for '%s': %s", field, resourceARN, err.Error())
+	if includeV4Metadata {
+		errResp, err := newErrorResponse(err, field, resourceARN)
+		if err == nil {
+			resp.Errors = append(resp.Errors, *errResp)
+		}
+	}
+}
+
+// newErrorResponse creates a new error response
+func newErrorResponse(err error, field, resourceARN string) (*ErrorResponse, error) {
+	if awsErr, ok := err.(awserr.Error); ok {
+		if reqErr, ok := err.(awserr.RequestFailure); ok {
+			errResp := &ErrorResponse{
+				ErrorField:   field,
+				ErrorCode:    awsErr.Code(),
+				ErrorMessage: awsErr.Message(),
+				StatusCode:   reqErr.StatusCode(),
+				RequestId:    reqErr.RequestID(),
+				ResourceARN:  resourceARN,
+			}
+			return errResp, nil
+		}
+	}
+
+	return nil, errors.Errorf("Unable to get an error response for resource '%s'", resourceARN)
```

Comment:
should we return something like below if the error is not `awserr.Error`?
```golang
return &ErrorResponse{
				ErrorField:   field,
				ErrorMessage: err.Error(),
				ResourceARN:  resourceARN,
			}
```

Reply:
Yes, we should. Updated the ErrorResponse to 
```
// newErrorResponse creates a new error response
func newErrorResponse(err error, field, resourceARN string) *ErrorResponse {
	errResp := &ErrorResponse{
		ErrorField:   field,
		ErrorMessage: err.Error(),
		ResourceARN:  resourceARN,
	}

	if awsErr, ok := err.(awserr.Error); ok {
		errResp.ErrorCode = awsErr.Code()
		errResp.ErrorMessage = awsErr.Message()
		if reqErr, ok := err.(awserr.RequestFailure); ok {
			errResp.StatusCode = reqErr.StatusCode()
			errResp.RequestId = reqErr.RequestID()
		}
	}

	return errResp
}
```

A non-awserr Error is tested with a mock Error, and responses will be shown as follows.
__TMDE v4__
```
{
   "Cluster":"default",
   "TaskARN":"xxx",
   ...
   "ContainerInstanceTags":{
      "tag_key":"tag_value"
   },
   "LaunchType":"EC2",
   "Errors":[
      {
         "ErrorField":"TaskTags",
         "ErrorMessage":"This is a non-awserr for testing: unable to find TaskTags for 'taskARN'",
         "ResourceARN":"taskARN"
      }
   ],
   "Containers":[{...}]
}
```
  
  __TMDE v3__
  ```
level=info time=xxx msg="V3 task metadata handler: writing response for task 'taskARN'" module=task_metadata_handler.go
level=error time=xxx msg="Task Metadata error: unable to get 'TaskTags' for 'taskARN': This is a non-awserr for testing: unable to find TaskTags for 'taskARN'" module=response.go
```
  
Thanks!